File: _staging/fix_encoding.js
const fs = require('fs');

const filepath = 'c:/Users/coman/Desktop/1.25.2026 christian website/index.html';

try {
    // Read the file as binary buffer
    const buffer = fs.readFileSync(filepath);

    // The current file content on disk is UTF-8 encoded, but the characters are wrong.
    // It contains the UTF-8 seq for 'Ù', which is 0xC3 0x99.
    // The original byte was 0xD9 (from Windows-1252 'Ù').
    // So we have the bytes C3 99 on disk.
    // We want 0xD9.
    // So we need to take the string "Ù...", encode it as Windows-1252 (or Latin1) to get the byte 0xD9.

    // Step 1: Read as UTF-8 string (which preserves the "wrong" characters)
    const wrongString = buffer.toString('utf8');

    // Step 2: Convert that string back to binary using 'binary' (latin1) encoding
    // This effectively takes the character code (0xD9) and puts it as a byte (0xD9).
    // Node's 'binary' is basically latin1.
    // However, if the file was saved as UTF-8 with BOM or something, we might have issues.
    // But let's assume standard UTF-8.

    // We want to map Char 'Ù' (0x00D9) -> Byte 0xD9.
    // Buffer.from(string, 'binary') does exactly this (truncates to lower 8 bits).
    const recoveredBuffer = Buffer.from(wrongString, 'binary');

    // Step 3: Now we have the original UTF-8 bytes (0xD9 0x85 for 'م') in the buffer.
    // We can just write this buffer back to disk.

    fs.writeFileSync(filepath, recoveredBuffer);
    console.log('Successfully fixed encoding for index.html');

} catch (err) {
    console.error('Error fixing encoding:', err);
    process.exit(1);
}

File: _maintenance_scripts/fix_encoding.py
import sys

def fix_encoding(filepath):
    try:
        # Read the file as UTF-8 (which it technically is now, but with wrong chars)
        with open(filepath, 'r', encoding='utf-8') as f:
            content = f.read()
        
        # The content is currently displayed as e.g. "Ù..."
        # We need to get the raw bytes that these characters represent in Latin-1/Windows-1252
        # and then interpret those bytes as UTF-8.
        
        # Trying Windows-1252 first as it's common on Windows
        try:
            # ex: The character 'Ù' (U+00D9) maps to byte 0xD9 in Windows-1252.
            # So if we encode to 'windows-1252', we get the original bytes back.
            raw_bytes = content.encode('windows-1252')
        except UnicodeEncodeError:
            # Fallback to latin-1 which maps 1:1 to first 256 unicode points
            raw_bytes = content.encode('latin-1')
            
        # Now decode these bytes as the intended UTF-8
        fixed_content = raw_bytes.decode('utf-8')
        
        with open(filepath, 'w', encoding='utf-8') as f:
            f.write(fixed_content)
            
        print(f"Successfully fixed encoding for {filepath}")
        
    except Exception as e:
        print(f"Error fixing encoding: {e}")
        sys.exit(1)

if __name__ == "__main__":
    fix_encoding(r'c:/Users/coman/Desktop/1.25.2026 christian website/index.html')
